columnphysics/icepack_wavefracspec.F90

```diff
@@ -720,6 +766,186 @@ subroutine get_fraclengths(X, eta, fraclengths, hbar)
 
       end subroutine get_fraclengths
 
+
+!===========================================================================
+!
+!
+! This routine contains the results of a pattern recognition network
+! (trained offline). The network classifies whether or not wave fracture occurs 
+! based on the 25-dim wave spectrum and ice thickness.
+! The output is an integer between 0 and 1.
+!
+!  authors: 2019 Lettie Roach, UW
+!                Chris Horvat, Brown University
+!
+
+      subroutine spwf_classifier(wave_spectrum, hbar, aice, &
+                                 spwf_classifier_out)
+
+
+      use icepack_parameters, only:  class_weight1, class_weight2, &
+                                    class_weight3, class_weight4, &
+                                    class_weight5, class_weight6
+
+
+      real (kind=dbl_kind), intent (in) :: &
+          hbar, aice  ! ice thickness (m), ice concentration
+
+      real (kind=dbl_kind), dimension (:), intent (in) :: &
+          wave_spectrum ! wave spectrum as a function of freq (m^s s)
+
+      real (kind=dbl_kind), intent(out) :: &
+          spwf_classifier_out
+
+
+      ! local variables
+
+      character(char_len_long) :: wave_class_file
+      
+!      real (kind=dbl_kind), dimension(27,100)  :: class_weight1
+!      real (kind=dbl_kind), dimension(100)     :: class_weight2
+!      real (kind=dbl_kind), dimension(100,100) :: class_weight3
+!      real (kind=dbl_kind), dimension(100)     :: class_weight4
+!      real (kind=dbl_kind), dimension(100,2)   :: class_weight5
+!      real (kind=dbl_kind), dimension(2)       :: class_weight6
+
+      real (kind=dbl_kind), dimension(27) :: input
+      real (kind=dbl_kind), dimension(100)    :: y1, y2
+      real (kind=dbl_kind), dimension(2)     :: y3
+
+      character(len=*), parameter :: subname = '(spwf_classifier)'
+
+      input(1:25) = wave_spectrum(1:25)
+      input(26)   = hbar
+      input(27)   = aice
+
+
+ 
+      y1 = MATMUL(input,class_weight1) + class_weight2
+      WHERE (y1 < c0) y1 = c0
```

Comment:
Rather than using WHERE, just use the intrinsic function
y1 = maxval(y1, c0)

Reply:
I just tried this but it gives a compile error:
/glade/u/home/lettier/CICE6/CICE/icepack/columnphysics/icepack_wavefracspec.F90(807): error #6362: The data types of the argument(s) are invalid.   [MAXVAL]
      y1 = MAXVAL(y1, c0)
I might be missing something very obvious but I don't see it!

Reply:
I'm mixing things up with another programming language.  I thought Fortran could do this but maxval isn't the right way, and I don't see another option.  Apologies for the confusion!